### PR TITLE
DROOLS-5749: [DMN Designer] Multiple DRDs support - When users open a DMN with duplicated information requirements, the marshaller doesn't fix it

### PR DESCRIPTION
**Pull request by karreiro**
**JIRA**: [DROOLS-5749](https://issues.redhat.com/browse/DROOLS-5749) - [DMN Designer] Multiple DRDs support - When users open a DMN with duplicated information requirements, the marshaller doesn't fix it

**Business Central**: [WAR file](https://www.dropbox.com/s/toe10dnah6sgduw/business-central-DROOLS-5749.war?dl=0)
**VS Code**: [plugin](https://www.dropbox.com/s/4auffyl3phrvvlb/vscode-plugin-DROOLS-5749.vsix?dl=0)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins do fdb.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/marshall/DMNMarshaller.java [hunk 1]
```diff
@@ -352,58 +365,41 @@ private void mergeNodeRequirements(final JSITDRGElement node,
...
-                        if (!existingKnowledgeSource.getAuthorityRequirement().contains(authorityRequirement)) {
-                            existingKnowledgeSource.addAuthorityRequirement(authorityRequirement);
-                        }
-                    });
+            existingKnowledgeSource.addAllAuthorityRequirement(nodeKnowledgeSource.getAuthorityRequirement().toArray(new JSITAuthorityRequirement[0]));
+            existingKnowledgeSource.setAuthorityRequirement(distinct(existingKnowledgeSource.getAuthorityRequirement()));
         }
     }
 
+    private <T extends JSITDMNElement> List<T> distinct(final List<T> list) {
```
Comment (jstastny-cz) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/marshall/DMNMarshaller.java:395]: Could we add here second argument and combine the lists like:
```
private <T extends JSITDMNElement> List<T> distinct(final List<T> list, final List<T> secondList) {
    List<T> toBeProcessed = new ArrayList(list);
    list.addAll(secondList);
    // the rest is the same, just process the combined list
    final Map<String, T> map = new HashMap<>();
    forEach(toBeProcessed, item -> {
        map.putIfAbsent(item.getId(), Js.uncheckedCast(item));
    });
    return new ArrayList<>(map.values());
}
```
The usage then wouldn't need to first to invoke `addAllAuthorityRequirement` and then retrospectively filtering and setting using `setAuthorityRequirement`.
Resulting in a single call for each requirement type
```
existingKnowledgeSource.setAuthorityRequirement(
    distinct(
        existingKnowledgeSource.getAuthorityRequirement(),
        nodeKnowledgeSource.getAuthorityRequirement()
    )
);
```
which is IMO nicer than
```
existingKnowledgeSource.addAllAuthorityRequirement(nodeKnowledgeSource.getAuthorityRequirement().toArray(new JSITAuthorityRequirement[0]));
existingKnowledgeSource.setAuthorityRequirement(distinct(existingKnowledgeSource.getAuthorityRequirement()));
```

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/marshall/DMNMarshaller.java (hunk 1 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/marshall/DMNMarshaller.java]: Nice enhancement. Thanks, @jstastny-cz!

**Comment by karreiro** [comment]
Thanks @jstastny-cz. Changes applied on new commit :-)

**Review by danielzhe — APPROVED** [review]

**Comment by jstastny-cz** [comment]
jenkins do fdb

**Review by jstastny-cz — APPROVED** [review]
LGTM

**Pull request merged**